### ISSUE TITLE
"run sketches on display" isn't working in 3.0a7

### ISSUE THREAD

**Issue by jddamas** [processing/processing#3264]
On 3.0a7 "run sketches on display" isn't working. I've set preferences window on second display and it's drawing on primary monitor with the dimensions of the second monitor. The sketch is below:

```
void init() {
  frame.removeNotify();
  frame.setUndecorated(true);
  frame.addNotify(); 
  super.init();
}

void setup() {
  size(displayWidth, displayHeight, P3D);
  smooth(8);
  frameRate(30);
  noCursor();
}
```

**Comment by benfry** [comment]
Will look into it. 

But in the meantime, where'd you get that code? That's really dangerous/unsupported stuff that should be completely unnecessary. If you use `size(displayWidth, displayHeight, P3D)`, 3.0a7 should detect that you're trying to run full screen and do all that automatically.

**Comment by jddamas** [comment]
I searched the web for a solution for making the window fullscreen without using present mode (on Yosemite, fullscreen apps running on second monitor make primary monitor go black, and I need the primary monitor for Max MSP). I saw that solution somewhere and it works as it removes the window frame. If I use 

void setup() {
  size(displayWidth, displayHeight, P3D);
  smooth(8);
  frameRate(30);
  noCursor();
}

only, the window won't go fullscreen (the frame is still there - see attached picture).
This is in a5. a7 second monitor isn't working like I said before.


![screen shot 2015-05-11 at 23 45 11 2](https://cloud.githubusercontent.com/assets/11235075/7576661/3dee0d8c-f838-11e4-8aba-af9defaae1f0.png)
